spec/models/expression_spec.rb

```diff
@@ -158,34 +158,34 @@
   end
 
   describe '#name_attr' do
-    let(:expected_name) do
-      'branch[conditionals_attributes][1][expressions_attributes][0][answer]'
-    end
+    let(:attributes) { %w[answer component operator field] }
 
-    it 'constucts the correct name attribute' do
-      expect(
-        expression.name_attr(
-          conditional_index: 1,
-          expression_index: 0,
-          attribute: 'answer'
-        )
-      ).to eq(expected_name)
+    it 'constructs the correct name attribute' do
+      attributes.map do |attribute|
+        expect(
+          expression.name_attr(
+            conditional_index: 1,
+            expression_index: 0,
+            attribute: attribute
+          )
+        ).to eq("branch[conditionals_attributes][1][expressions_attributes][0][#{attribute}]")
+      end
     end
   end
 
   describe '#id_attr' do
-    let(:expected_id) do
-      'branch_conditionals_attributes_0_expressions_attributes_1_operator'
-    end
+    let(:attributes) { %w[component operator field] }
 
     it 'constructs the correct id attribute' do
-      expect(
-        expression.id_attr(
-          conditional_index: 0,
-          expression_index: 1,
-          attribute: 'operator'
-        )
-      ).to eq(expected_id)
+      attributes.map do |attribute|
+        expect(
+          expression.id_attr(
+            conditional_index: 0,
+            expression_index: 1,
+            attribute: attribute
+          )
+        ).to eq("branch_conditionals_attributes_0_expressions_attributes_1_#{attribute}")
+      end
```

Comment:
i kinda feel like both these test updates aren't necessary really. the controller.js changes don't interact with the name_attr or id_attr methods on the expression object.

if we dont' have JS tests to update then the best thing would be acceptance tests that checks something like:

- no errors are thrown
- the operator and answer chosen remain consistent across more than one conditional

Reply:
Acceptance tests added!